File: src/main/java/io/nightfis/createstellarexoduscore/filter/IsPlayerIdFilter.java
package io.nightfis.createstellarexoduscore.filter;

import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.player.Player;

public class IsPlayerIdFilter extends TargetFilter {

    private static final String ID = "IsPlayerIdFilter";

    @Override
    public String getId() {
        return ID;
    }

    @Override
    public boolean match(LivingEntity entity, String arg) {
        return entity instanceof Player player && player.getName().getString().equals(arg);
    }

    @Override
    public boolean needArg() {
        return true;
    }

    @Override
    public Component title() {
        return Component.translatable("screen.createstellarexoduscore.auto_aim_turret.filter.is_player_id");
    }

    @Override
    public String tooltip(String arg) {
        return arg;
    }
}

File: src/main/kotlin/io/nightfis/createstellarexoduscore/filter/IsPlayerIdFilter.kt
package io.nightfis.createstellarexoduscore.filter

import net.minecraft.network.chat.Component
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.player.Player

class IsPlayerIdFilter : TargetFilter() {

    override fun getId(): String = ID

    override fun match(entity: LivingEntity, arg: String): Boolean {
        return entity is Player && entity.name.string == arg
    }

    override fun needArg(): Boolean = true

    override fun title(): Component {
        return Component.translatable("screen.create_stellar_exodus_core.auto_aim_turret.filter.is_player_id")
    }

    override fun tooltip(arg: String): String = arg

    companion object {
        private const val ID = "IsPlayerIdFilter"
    }
}
